Update request-tokens.md (#325)
contao/docs: docs/dev/framework/request-tokens.md

Providing CSRF protection for users that are not authenticated against the app does not make any sense. So if you
visit a regular Contao page with a form placed on it, you will not necessarily see any cookies being set in your
browser. Only if you are authenticated in such a way that the browser will automatically send authentication information

> along without any user interaction (logged in via `Cookie` or basic authentication  via `Authorization` headers) or the
> request has any other cookies, CSRF protection is required. So don't get fooled by the cookies not being present all the time, Contao is actually very smart about them to improve HTTP cache hits.

Reviewer: ```suggestion
along without any user interaction (e.g. any cookies or basic authentication via `Authorization` headers), CSRF
protection is required. So don't get fooled by the cookies not being present all the time, Contao is actually very smart
about them to improve HTTP cache hits.
```

> By default, Contao protects all `POST` requests except Ajax requests coming from Contao routes. That means, routes that either do have

Reviewer: Hm, this I find confusing too. Why is _any_ AJAX request not checked? Shouldn't any `POST` request being checked, regardless of whether it is identified as an AJAX request or not?

Reviewer 2: Ajax requests cannot be used for CSRF attacks, because they are secured against cross domain requests by the browser.

Reviewer 2: To be more precise: We check for AJAX requests by validating that the HTTP header `X-Requested-With` was set to `XMLHttpRequest`, see https://github.com/symfony/symfony/blob/e1a522b47077b7673090912e33469ac9f0b72a67/src/Symfony/Component/HttpFoundation/Request.php#L1712

Setting custom HTTP headers is only allowed for AJAX requests on the same origin and doesn’t work cross-domain, except if it was explicitly allowed by a CORS preflight request. For details see <https://cheatsheetseries.owasp.org/cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.html#use-of-custom-request-headers>

Reviewer 3: I think in this PR we should just omit any explanation and just write we don't check request tokens for ajax requests. We can include the explanation for it in an info notice. I'll do a PR if @Mynyx doesn't have the energy anymore :)

Author: @Toflar A PR would be welcome. :)

Reviewer 3: Sure, I'll do that once this PR is ready. Please remove any explanation and incorporate other reviews so we can finish this :)
